tools/history_tools.py

```
@@ -33,15 +46,19 @@ async def read_chat_history(self, limit: int = 2000) -> str:
+            # Use DB directly with error handling
+            try:
+                db_messages = await get_messages(channel_id, limit)
+                if not db_messages:
+                    return "No history found in database."
+                    
+                formatted = []
+                for m in db_messages:
+                    formatted.append(f"{m['timestamp_str']} {m['author_name']}({m['author_id']}): {m['content']}")
+                return "\n".join(formatted)
+            except Exception as e:
+                logger.error(f"[HistoryTools] Database error for channel {channel_id}: {e}", exc_info=True)
```

Ruff (0.15.2)

[warning] Logging `.exception(...)` should be used instead of `.error(..., exc_info=True)`

(G201)